src/Confluent.Kafka/KafkaException.cs

```diff
@@ -23,12 +23,19 @@ namespace Confluent.Kafka
 {
     public class KafkaException : Exception
     {
-        public KafkaException(Error error, string message)
-            : base(message)
+        private Error _error;
```

Comment:
Yeah. There are recommendations by MS (but they themselves don't follow them in their code)

I mean: 

```
    public class KafkaException : Exception
    {
        public KafkaException(Error error)
            : base(error.ToString())
        {
            Error = error;
        }

        public KafkaException(ErrorCode code)
            : base(Error.ErrorCode2String(code))
        {
            Error = new Error(code);
        }

        public Error Error { get; }
    }
```